File: Table.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.IO;

namespace DataBaseManager
{
    public class Table
    {
        public string Error = "";
        public string Msg = "";
        public string SearchMsg = "";


        public string Name { get; private set; }
        public string DbFilePath { get; private set; }
        
        public Dictionary<string, Column> Columns { get; private set; }

        public string PrimaryKey { get; private set; }
        
        public List<List<string>> Data { get; private set; }


        public Table(string name, string dbFilePath)
        {
            Name = name;
            DbFilePath = dbFilePath;
        }


        //列情報（Columns）を設定
        public bool SetColumns(string[] names, string[] types, bool[] isNullables, string[] alias, string primaryKey)
        {
            Columns = new Dictionary<string, Column>();

            //各要素の数チェック
            if ((names.Count() != types.Count()) || (names.Count() != isNullables.Count()) || (names.Count() != alias.Count()))
            {
                Error = "列を設定するための各要素の数が一致しません。\r\n";
                return false;
            }
            //typesチェック
            foreach (string type in types)
            {
                if (!(new List<string> { "int","string","date" }).Contains(type))
                {
                    Error = $"型名:{type}は有効でない値です。\r\n";
                    return false;
                }
            }
            //primaryKeyチェック
            if (!names.Contains(primaryKey))
            {
                Error = $"プライマリーキー:{primaryKey}は有効でない値です。\r\n";
                return false;
            }

            //設定
            try
            {
                for (int i = 0; i < names.Count(); i++)
                {
                    Columns[names[i]] = new Column(names[i], types[i], isNullables[i], alias[i], i);
                }

                PrimaryKey = primaryKey;
            }
            catch (Exception ex)
            {
                Error = $"列設定に失敗しました。\r\n{ex.ToString()}\r\n";
                return false;
            }
            return true;
        }


        public bool SetData()
        {            
            Data = new List<List<string>>();

            if (!File.Exists(DbFilePath))
            {
                return false;
            }

            using (StreamReader sr = new StreamReader(DbFilePath))
            {
                string line = "";
                while ((line = sr.ReadLine()) != null)
                {
                    var buff = line.Split(',');
                    Data.Add(buff.ToList());
                }
            }
            return true;
        }
    }
}

File: table.py
import column

class Table:
    
    def __init__(self,name,db_file_path):
        self.error = ""
        self.msg = ""
        
        self.name = name
        self.db_file_path = db_file_path
        
        self.columns = {}
        self.data = []
        
        
    #列情報を設定する
    def set_columns(self,names,types,is_nullables,aliases,primary_key):
        #各要素の数チェック
        if len(names) != len(types) or len(names) != len(isNullables) or len(names) != len(alias):
            self.error = "列を設定するための各要素の数が一致しません。\n"
            return False
        #typesチェック
        for tp in types:
            if tp not in ["int","string","date"]:
                self.error = f"型名:{tp}は有効でない値です。\n"
                return False
        #pkチェック
        if primary_key not in names:
            self.error = f"プライマリーキー:{primary_key}は有効でない値です。\n"
            return False
        
        #列設定
        //設定
        try:
            for i in range(len(names)):
                columns[names[i]] = column.Column(names[i],types[i],is_nullables[i],aliases[i],i)                
            return True
        except Exception as ex:
            self.error = "列の設定に失敗しました。\n"+str(ex)+"\n"
            return False
    
    #データを設定する   
    def set_data():
        if os.path.exists(self.db_file_path):
            return False
        
        try:
            with open(self.db_file_path,"r",encoding="utf-8") as f:
                for line in f:
                    data.append(line.replace("\n","").split(","))
        except Exception as ex:
            self.error = f"データファイル:{self.db_file_path}の読み込みに失敗しました。\n{ex}\n"
            return False
